lib/ex_aws/operation/query.ex

```diff
@@ -10,6 +10,7 @@ defmodule ExAws.Operation.Query do
 
   defstruct path: "/",
             params: %{},
+            content_encoding: "identity",
```

Comment:
I think it'd be better to just provide a generic `headers` key here.

Reply:
And look for `content_encoding` in these headers in `perform` to decide whether we should compress `data` or not?
If it's ok for you then I can do that.

Reply:
@benwilson512 could you please answer my question above? I'm already using this code for more than a month in production and I would be glad to update this PR so it could be merged.